free-2.md

```diff
@@ -11,41 +11,85 @@ _What need is met by your idea? Who are you helping? What is that person's
+line numbers, the difficult debugging, and other problems. So, I figured
+that one option would be to rewrite HMMM, fixing these problems._
 
 ### Domain
 _Describe the project's domain in five words._
 
+_Simplified assembly for learning CS_
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+_The user will interact with this language by writing a pseudo-assembly
+code. The programs will look like lines of assembly code. However,
+HMMM only knows about a handful of commands. So, each line should be
+one of these commands. This is the right way to interact with the domain
+because it will let students learn the ideas of assembly without being
+overwhlemed by actual assembly and all the "gotchas" that come with that._
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+_When a program runs, it will take the users input and run it based on the
+rules that we have written. It will interact with the user because the user
+will tell the language the program that they want to run and the language
+will give the user the output. It could get errors that it cannot run the
+program because it does not know a command or the command does not have the
+right inputs. Hopefully, it would be able to comminicate to the user the
+lines that that occured and which of these problems it is._
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+_It should be easy to write simple programs in assembly. It would be difficult
+yet possible to write recursion in assembly. It would be impossible to create
+data structures in this language._
 
 ### Related work
 _Are there any other DSLs in this domain? If not, describe how you know there
 aren't and conjecture why not. If so, describe them and provide links. How well 
 do they address the need? Are there any particularly admirable qualities of the
 language? Are there parts of the language you think could be improved?_
 
+_Right now, there is the current implementation of HMMM. It can be found here:
+https://www.cs.hmc.edu/~cs5grad/cs5/hmmm/documentation/documentation.html. I
+really admire that it keeps it simple. Right now, there are only 23 instructions,
+which makes it easy for students to have a reference guide of what each
+instruction is. I think the languages' debugging could be better. Right now,
+students must either dump memory or print registers. I would be interested in
+seeing if there are better ways to debug (not print out the full memory or 
+have an easier way to make automatic breakpoints). I would also be interested
+in how possible it is to remove the line numbers._
```

Comment:
I know Jacob Roth has a python script that allows you to use variables for line numbers, so you can refer to them in their code. The script will then replace the variables with line numbers so you wouldn't need to adjust line numbers as you write code. As an aside, if I recall correctly, he was told not to distribute the script because it would be too much help for the CS 5 assignments.

I think an alternative method for the line number problem is if you let each line be a movable block, kind of like Scratch. You refer to these blocks instead of using line numbers to do loops.